### ISSUE TITLE
Investigate retries for kubectl commands

### ISSUE THREAD

**Issue by Walther** [garden-io/garden#4591]
Currently, we can sometimes see errors like

```
2023-06-09T09:09:25.3573989Z error: error when retrieving current configuration of:
2023-06-09T09:09:25.3574834Z Resource: "apps/v1, Resource=deployments", GroupVersionKind: "apps/v1, Kind=Deployment"
2023-06-09T09:09:25.3634918Z Name: "redacted", Namespace: "redacted"
2023-06-09T09:09:25.3636527Z from server for: "STDIN": Get "redacted": dial tcp redacted_ip:443: i/o timeout
```

We should investigate whether we have logic set up for retries - either in our own code, or by `kubectl`'s own features, possibly depending on `kubectl` version as well, etc

**Comment by vvagaytsev** [comment]
It looks like there is no retry machinery available for `kubectl`: https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands

**Comment by vvagaytsev** [comment]
We use `kubectl` in a few places. Maybe we should wrap the most frequently used calls into our own retry machinery.

**Comment by vvagaytsev** [comment]
In #4731 we implemented retries for config retrieval over `kubectl`. The improvement will be released in `0.13.6`.

**Comment by Walther** [comment]
Closing as addressed in https://github.com/garden-io/garden/pull/4731
